rust/chains/hyperlane-ethereum/src/rpc_clients/trait_builder.rs

```diff
@@ -234,15 +244,23 @@ pub trait BuildableWithProvider {
 async fn wrap_with_signer<M: Middleware>(
     provider: M,
     signer: Signers,
-) -> Result<SignerMiddleware<NonceManagerMiddleware<M>, Signers>, M::Error> {
+) -> Result<SignerMiddleware<M, Signers>, M::Error> {
     let provider_chain_id = provider.get_chainid().await?;
     let signer = ethers::signers::Signer::with_chain_id(signer, provider_chain_id.as_u64());
 
-    let address = ethers::prelude::Signer::address(&signer);
-    let provider = NonceManagerMiddleware::new(provider, address);
+    Ok(SignerMiddleware::new(provider, signer))
+}
+
+async fn wrap_with_nonce_manager<M: Middleware>(
+    provider: M,
+    signer: Signers,
+) -> Result<NonceManagerMiddleware<M>, M::Error> {
+    let provider_chain_id = provider.get_chainid().await?;
+    let signer = ethers::signers::Signer::with_chain_id(signer, provider_chain_id.as_u64());
```

Comment:
I think we can remove this now and just pass in `signer_address: &Address` into this function. Bc we just create the signer here only for the address, and the signer is actually used in `wrap_with_signer`